lib/posix/options/posix_internal.h

```diff
@@ -29,6 +29,7 @@ struct posix_thread_attr {
 	uint16_t guardsize : CONFIG_POSIX_PTHREAD_ATTR_GUARDSIZE_BITS;
 	int8_t priority;
 	uint8_t schedpolicy: 2;
+	uint8_t contentionscope: 2;
```

Comment:
It looks like there are might only be 2 possible values that this can take. Is that right?

SYSTEM and PROCESS?

If that's the case, then the number of bits required to encode it should be `log2(2) = 1`

Otherwise, LGTM - a couple more tests, and it should be ok. 
```suggestion
	uint8_t contentionscope: 1;
```